api/youtube_downloader.py

```
@@ -166,22 +167,27 @@ def __init__(self, video_url: str):
+                r'(?:embed\/|youtu\.be\/)([0-9A-Za-z_-]{11})' 
+            ]
+            for pattern in patterns:
+                match = re.search(pattern, url)
+                if match:
+                    video_id = match.group(1)
+                    logger.debug(f"Extracted video ID: {video_id}")
+                    return video_id
+            return None
+        except Exception as e:
+            detailed_error = handle_error(e, {"url": url, "function": "_extract_video_id"}, logger)
+            raise detailed_error.error
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)